### ISSUE TITLE
Error handling for 'Upcoming' videos in playlists

### ISSUE THREAD

**Issue by DavidUllauri** [sharatsachin/ytplaylist-len#38]
When trying to find the length of a playlist that contains an 'Upcoming' video, the site returns a status 500 Internal Server Error.

This occurs because `duration` in `contentDetails` is undefined for 'Upcoming' videos.

**Comment by sanketkamta** [comment]
Can you provide me with any such link?

**Comment by sanketkamta** [comment]
The issue has been fixed here - https://youtube-playlist-length-e8xu.onrender.com/

**Comment by sharatsachin** [comment]
@DavidUllauri Please check if this is fixed as per the latest deployment.

**Comment by DavidUllauri** [comment]
Thank you for addressing the issue! I've tested it again, and while the error is now handled more gracefully with the error message `Error: 'Video' object has no attribute 'duration'`, the playlist length is still not returned when it contains an 'Upcoming' video. It would be great if the playlist length could be calculated while excluding upcoming videos.

This is the link I used for testing: https://youtube.com/playlist?list=PLD_vBJjpCwJvP9F9CeDRiLs08a3ldTpW5&si=-8bwNRud1niknuxm
